framework/common/src/main/java/edu/kit/kastel/mcse/ardoco/core/api/models/arcotl/code/CodeItem.java

```diff
@@ -43,7 +43,6 @@ protected CodeItem(CodeItemRepository codeItemRepository, String name) {
     }
 
     void registerCurrentCodeItemRepository(CodeItemRepository codeItemRepository) {
-        codeItemRepository.addCodeItem(this);
```

Comment:
I removed this during profiling, because the function is only called in the CodeItemRepository init on the values of the repository. It is never used in a context where a CodeItem is unregistered. The Code SAM Tlr stuff was really slow, mainly because the CodeItemRepository kept being reinitialized, which is very costly when you consider that it is sorted and its size. So I made sure that it would properly be initialized only once and that unnecessary add calls are removed
Long story short: Its intentional, not breaking and increases performance

Reply:
Did you also tested the recreation of code models or only the loading?

Reply:
Seems to be safe :)